tendrl/commons/event.py

```diff
@@ -0,0 +1,17 @@
+import _socket
+from tendrl.commons.message import Message
+
+
+PATH = "/etc/tendrl/node_agent_socket"
+
+
+class Event(object):
+    def __init__(self, message):
+        self.sock = _socket.socket(_socket.AF_UNIX, _socket.SOCK_STREAM)
+        self.write(message)
+
+    def write(self, message):
+        self.sock.connect(PATH)
```

Comment:
how user gets to know?

For example, in case of alerting, the threshold detection happens @ collectd, the plugin invokes the Message framework below and then the framework's attempt to open a socket connection to node-agent exposed socket fails.
In this case the plugin gets to know about the exception and the maximum it can do is log the failure to collectd logs.But how can that be known by the user I am asking this more with this perspective, if its an error/exception happening internal to collectd, probably user going to collectd logs might seem logical but if its an issue very much specific to processes internal to tendrl, I would like there to be a way for user to be aware of it from the tendrl logging framework itself rather than from integration component logs.

Also consider another example, the ceph and gluster integrations will be responsible for detecting state changes of entities managed by them, consider if this connect fails, then what happens how can it be made known to user?

Or modules would terminate at the very init if the socket creation @ node-agent and connection @ other modules end failed is my assumption correct? In such cases also, how is it planned to log/notify error in such cases to user?

With socket path hard-coded and communication being local within the node itslef, I am not sure about likelihood of such a failure with packaging doing the necessary permissions to the sock file and ensuring existence of socket file.
But, I would like to take opinions from @Tendrl/tendrl-core here